File: abschnitt-1/lösung/Workshop.WebApi/Controllers/EchoController.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace Workshop.WebApi.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class EchoController : ControllerBase
    {

        [HttpGet("simple")]
        public ActionResult<string> Simple() {
            return Ok("Hallo Welt");
        }

        [HttpGet("parameter")]
        public ActionResult<string> Parameter([FromQuery] string msg) {
            return Ok(msg);
        }

        [HttpGet("validate/{msg}")]
        public ActionResult<string> Validate([FromRoute] string msg = null) {
            if (string.IsNullOrWhiteSpace(msg)) {
                return BadRequest();
            }
            return Ok(msg);
        }
    }
}

File: abschnitt-2/lösung/workshop-node-api/src/echo/echo.controller.ts
import { BadRequestException, Controller, Get, Param, Query, Req } from '@nestjs/common';

@Controller('/api/echo')
export class EchoController {
    
    @Get('simple')
    simple(): string {
        return "Hallo Welt";
    }

    @Get('parameter')
    parameter(@Query() msg: string): string {
        return msg;
    }

    @Get('validate/:msg?')
    validate(@Param('msg') msg: string | undefined = undefined): string {
        if (!msg) {
            throw new BadRequestException(msg, "msg undefined");
        }
        return msg;
    }
}
